pkg/template/apis/template/constants.go

```diff
@@ -56,4 +56,9 @@ const (
 	// Base64ExposeAnnotationPrefix is as ExposeAnnotationPrefix, except that
 	// any []byte values returned are base64 encoded.
 	Base64ExposeAnnotationPrefix = "template.openshift.io/base64-expose-"
+
+	// WaitForReadyAnnotation indicates that the TemplateInstance controller
+	// should wait for the object to be ready before reporting the template
+	// instantiation complete.
+	WaitForReadyAnnotation = "template.alpha.openshift.io/wait-for-ready"
```

Comment:
if we're going to have the annotation be necessary than yeah, i think we'd have to put it on the templates we control (and ideally, on the middleware templates too) to be good citizens.

As for documentation, i think we'd document it in the "writing a template" guide, but we'd need to note it as being tech preview/subject to change.  Which of course means we're going to make template authors unhappy when we suddenly stop respecting this value (ie not waiting for completion).

Which, while "legitimate" under the tech preview umbrella, feels like we're just setting ourselves up to make people unhappy.

So i'd still sort of prefer that waiting just be a thing we do, if we know how to wait for the resource.